src/main/java/seedu/clinic/logic/commands/EditCommand.java

```diff
@@ -305,7 +330,13 @@ public void setRemark(Remark remark) {
          * A defensive copy of {@code products} is used internally.
          */
         public void setProducts(Set<Product> products) {
-            this.products = (products != null) ? new HashSet<>(products) : null;
+            if (products == null) {
+                this.products = null;
+            } else if (products.isEmpty()) {
```

Comment:
Is this meant to be an additional level of check?

Reply:
Yes, because of the concern about NullPointerException